docs/style-guide.adoc

```diff
@@ -0,0 +1,455 @@
+RETURN own.since
+----
+
+. Use camel case, starting with a lower case character, for:
+.. functions
+.. properties
+.. variables
+.. parameters
++
+.Bad
+[source, cypher]
+----
+CREATE (N {Prop: 0})
+WITH RAND() AS Rand, {param} AS MAP
+RETURN Rand, MAP.property_key, Count(N)
+----
++
+.Good
+[source, cypher]
+----
+CREATE (n {prop: 0})
+WITH rand() AS rand, {param} AS map
+RETURN rand, map.propertyKey, count(n)
+----
+
+. Use single quotes (Unicode character U+0027: ') for literal string values.
++
+.Bad
+[source, cypher]
+----
+RETURN "Cypher"
+----
++
+.Good
+[source, cypher]
+----
+RETURN 'Cypher'
+----
+
+. *Spacing*
+.. For literal maps:
+... No space between opening brace and first key
+... No space between key and colon
+... One space between colon and value
+... No space between value and comma
+... One space between comma and next key
+... No space between last value and closing brace
++
+.Bad
+[source, cypher]
+----
+WITH { key1 :'value' ,key2  :  42 } AS map
+RETURN map
+----
++
+.Good
+[source, cypher]
+----
+WITH {key1: 'value', key2: 42} AS map
+RETURN map
+----
+
+.. No padding space for parameters.
++
+.Bad
+[source, cypher]
+----
+RETURN { param }
+----
++
+.Good
+[source, cypher]
+----
+RETURN {param}
+----
+
+.. One space between label/type predicates and property predicates in patterns.
++
+.Bad
+[source, cypher]
+----
+MATCH (p:Person{property: -1})-[:KNOWS   {since: 2016}]->()
+RETURN p.name
+----
++
+.Good
+[source, cypher]
+----
+MATCH (p:Person {property: -1})-[:KNOWS {since: 2016}]->()
+RETURN p.name
+----
+
+.. No space in patterns.
++
+.Bad
+[source, cypher]
+----
+MATCH (:Person) --> (:Vehicle)
+RETURN count(*)
+----
++
+.Good
+[source, cypher]
+----
+MATCH (:Person)-->(:Vehicle)
+RETURN count(*)
+----
+
+.. Use wrapping space around operators.
++
+.Bad
+[source, cypher]
+----
+MATCH p=(s)-->(e)
+WHERE s.name<>e.name
+RETURN length(p)
+----
++
+.Good
+[source, cypher]
+----
+MATCH p = (s)-->(e)
+WHERE s.name <> e.name
+RETURN length(p)
+----
+
+.. No space in label predicates.
++
+.Bad
+[source, cypher]
+----
+MATCH (person    : Person  :  Owner  )
+RETURN person.name
+----
++
+.Good
+[source, cypher]
+----
+MATCH (person:Person:Owner)
+RETURN person.name
+----
+
+.. Use space after each comma in lists and enumerations.
++
+.Bad
+[source, cypher]
+----
+MATCH (),()
+WITH ['a','b',3.14] AS list
+RETURN list,2,3,4
+----
++
+.Good
+[source, cypher]
+----
+MATCH (), ()
+WITH ['a', 'b', 3.14] AS list
+RETURN list, 2, 3, 4
+----
+
+. When patterns wrap lines, break after arrows, not before.
++
+.Bad
+[source, cypher]
+----
+MATCH (:Person)-->(vehicle:Car)-->(:Company)
+      <--(:Country)
+RETURN count(vehicle)
+----
++
+.Good
+[source, cypher]
+----
+MATCH (:Person)-->(vehicle:Car)-->(:Company)<--
+      (:Country)
+RETURN count(vehicle)
+----
+
+. Surround node patterns with parentheses.
+.. This only concerns old Cypher code, as it is no longer legal Cypher to omit parentheses.
```

Comment:
-> "This only concerns old Cypher code, as omitting parentheses is no longer legal in Cypher."